compiler/src/dmd/expression.d

```diff
@@ -7398,3 +7398,134 @@ private enum EbinaryAssign =
         EXP.leftShiftAssign, EXP.rightShiftAssign, EXP.unsignedRightShiftAssign,
         EXP.concatenateAssign, EXP.concatenateElemAssign, EXP.concatenateDcharAssign,
     ];
+
+/// Given a member of the EXP enum, get the class instance size of the corresponding Expression class.
+/// Needed because the classes are `extern(C++)`
+private immutable ubyte[EXP.max+1] expSize = [
```

Comment:
Note: this _isn't_ thread-local because `immutable` - so no need for explicit `__gshared`.

Reply:
Where do you see `__gshared`?